File: Inheritance(C++)/Multi_Level.cpp
#include <iostream>

using namespace std;

class Father
{
    protected:
        int age;

    public:
        void getage()
        {
            age = 75;
        }
};

class Son : public Father // Son class inherited from Father
{
    protected:
        int S_age;

    public:
        void getage_S()
        {
            S_age = 45;
        }
};

class GrandSon : public Son  // Grandson class inherited from Son
{
    protected:
        int GS_age;

    public:
        void getage_GS()
        {
            GS_age = 15;
        }
};

int main()
{
	GrandSon obj;

	obj.getage();
	obj.getage_S();
	obj.getage_GS();

	cout << "Father's age :" << obj.age;
    cout << "\nSon's age :" << obj.S_age;
	cout << "\nGrandson's age :" << obj.GS_age;

    return 0;
}

/* Output

Father's age : 75
Son's age : 45
Grandson's age : 15

*/

File: Inheritance(Java)/Multi_Level.java
class Father
{
    int age;

    public void setage()
    {
        age = 75;
    }

    public int getage()
    {
        return age;
    }
}

class Son extends Father
{
    int S_age;

    public void setage_S()
    {
        S_age = 45;
    }

    public int getage_S()
    {
        return S_age;
    }
}

class GrandSon extends Son
{
    int GS_age;

    public void setage_GS()
    {
        GS_age = 15;
    }

    public int getage_GS()
    {
        return GS_age;
    }
}

class Multi_Level
{
    public static void main(String args[])
    {
        GrandSon obj = new GrandSon();

        obj.setage();
    	obj.setage_S();
    	obj.setage_GS();

        System.out.println("Father's age : " + obj.getage());
        System.out.println("Son's age : " + obj.getage_S());
    	System.out.println("Grandson's age : " + obj.getage_GS());
    }
}

/* Output

Father's age : 75
Son's age : 45
Grandson's age : 15

*/
